Add 2 small test cases for Board.add_stone

Gr0minet opened the pull request:
- add Board.compute_groups_from_scratch
 - Board.display is now Board.as_string for a smoother usage

Gr0minet committed:
Add 2 small test cases for Board.add_stone

 - add Board.compute_groups_from_scratch
 - Board.display is now Board.as_string for a smoother usage

m09 commented on shared/src/main/scala/govariants/Board.scala:
```diff
@@ -134,6 +137,45 @@ class Board(val size: Int) {
   }
 
+  def compute_groups_from_scratch() = {
+    val visited: HashMap[Intersection, Boolean] = HashMap()
```
Should probably be a set of intersections instead of a map (set membership will be the boolean value we store here).

m09 commented on shared/src/main/scala/govariants/Board.scala:
```diff
@@ -20,25 +21,27 @@ class Board(val size: Int) {
 
-  def display() = {
-    print("  ")
+  def as_string(): String = {
```
It's great to switch to a pure function. We can probably even switch to `@override def toString`.

m09 reviewed (approved):
We even have tests now :frog: I added the usual few nitpicks.

Gr0minet committed:
Minor changes for Github PR#4

 - 'visited' from Board.compute_groups_from_scratch is now a Set instead
 of a HashMap
 - Board.as_string is now Board.toString (override)
